src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -5,23 +5,26 @@ import { Address, LegacyAddress } from '../../../address';
 import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
 import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType, StandardError } from '../../../common/error/errors';
 import { PaymentMethod } from '../../../payment';
-import { BraintreeError, BraintreePaypalCheckout, BraintreeShippingAddressOverride, BraintreeSDKCreator, BraintreeTokenizePayload } from '../../../payment/strategies/braintree';
-import { PaypalAuthorizeData, PaypalScriptLoader } from '../../../payment/strategies/paypal';
+import { BraintreeError, BraintreePaypalCheckout, BraintreeShippingAddressOverride, BraintreeSDKCreator, BraintreeTokenizePayload, RenderButtonsData } from '../../../payment/strategies/braintree';
+import { PaypalAuthorizeData, PaypalHostWindow } from '../../../payment/strategies/paypal';
 import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
 import CheckoutButtonStrategy from '../checkout-button-strategy';
 
 export default class BraintreePaypalButtonStrategy implements CheckoutButtonStrategy {
     private _paypalCheckout?: BraintreePaypalCheckout;
     private _paymentMethod?: PaymentMethod;
+    private _window?: PaypalHostWindow;
 
     constructor(
         private _store: CheckoutStore,
         private _checkoutActionCreator: CheckoutActionCreator,
         private _braintreeSDKCreator: BraintreeSDKCreator,
-        private _paypalScriptLoader: PaypalScriptLoader,
         private _formPoster: FormPoster,
-        private _offerCredit: boolean = false
-    ) {}
+        private _offerCredit: boolean = false,
+        private _renderButtonsData?: RenderButtonsData
+    ) {
+        this._window = window;
```

Comment:
rather than accessing the windows, can't you keep what you had before? 
```
 private _paypalScriptLoader: PaypalScriptLoader,
```

or if you don't need the loader, can't you pass instead the `window.paypal` object in the constructor? This way you will be able to properly unit test this.